trunk/source/Library-ModelCheckerUtils/src/de/uni_freiburg/informatik/ultimate/lib/modelcheckerutils/cfg/ProcedureMultiplier.java

```diff
@@ -271,28 +260,19 @@ private ForkSmtArguments copyForkSmtArguments(final ForkSmtArguments forkSmtArgu
 		return new ForkSmtArguments(newThreadIdArguments, newProcedureArguments);
 	}
 
-	private JoinSmtArguments copyJoinSmtArguments(final JoinSmtArguments joinSmtArguments,
+	private static JoinSmtArguments copyJoinSmtArguments(final JoinSmtArguments joinSmtArguments,
 			final Map<ILocalProgramVar, ILocalProgramVar> map, final ManagedScript managedScript) {
 		final Map<Term, Term> defaultVariableMapping = constructDefaultVariableMapping(map);
 		final MultiTermResult newThreadIdArguments =
 				copyMultiTermResult(joinSmtArguments.getThreadIdArguments(), defaultVariableMapping, managedScript);
-		final List<IProgramVar> newAssignmentLhs =
-				joinSmtArguments.getAssignmentLhs().stream().map(x -> getNew(map, x)).collect(Collectors.toList());
+		final List<IProgramVar> newAssignmentLhs = joinSmtArguments.getAssignmentLhs().stream()
+				.map(x -> x instanceof ILocalProgramVar ? map.get(x) : x).collect(Collectors.toList());
 		return new JoinSmtArguments(newThreadIdArguments, newAssignmentLhs);
 	}
 
-	private static IProgramVar getNew(final Map<ILocalProgramVar, ILocalProgramVar> mapOld2New,
-			final IProgramVar variable) {
-		if (variable instanceof ILocalProgramVar) {
-			return mapOld2New.get(variable);
-		}
-		assert variable.isGlobal() : "Variable is neither local nor global";

```

Comment:
I think I added this assertion once (after introducing the method to fix a bug when a join statement assigned the return value to a global variable). I don't mind removing it, but (real question) couldn't we use `isGlobal` rather than `instanceof` to make the distinction?

Reply:
Yes, we can use `!variable.isGlobal()` instead of `variable instanceof ILocalProgramVar`

Reply:
Okay, should I replace this in the stream-expression then? Should not change anything

Reply:
:+1: I think it would be good to replace this